react/features/recording/components/RecordingLabel.web.js

```diff
@@ -5,243 +5,35 @@ import { connect } from 'react-redux';
-     */
-    hidden: boolean
-};
-
 /**
  * Implements a React {@link Component} which displays the current state of
  * conference recording.
  *
  * @extends {Component}
  */
-class RecordingLabel extends AbstractRecordingLabel<Props, State> {
-    _autohideTimeout: TimeoutID;
-
-    state = {
-        hidden: false
-    };
-
-    static defaultProps = {
-        session: {}
-    };
-
-    /**
-     * Sets a timeout to automatically hide the {@link RecordingLabel} if the
-     * recording session started as failed.
-     *
-     * @inheritdoc
-     */
-    componentDidMount() {
-        if (this.props.session.status === JitsiRecordingConstants.status.OFF) {
-            this._setHideTimeout();
-        }
-    }
-
-    /**
-     * Sets a timeout to automatically hide {the @link RecordingLabel} if it has
-     * transitioned to off.
-     *
-     * @inheritdoc
-     */
-    componentWillReceiveProps(nextProps) {
-        const { status } = this.props.session;
-        const nextStatus = nextProps.session.status;
-
-        if (status !== JitsiRecordingConstants.status.OFF
-            && nextStatus === JitsiRecordingConstants.status.OFF) {
-            this._setHideTimeout();
-        }
-    }
-
+class RecordingLabel extends AbstractRecordingLabel<Props> {
     /**
-     * Clears the timeout for automatically hiding the {@link RecordingLabel}.
+     * Renders the platform specific label component.
      *
      * @inheritdoc
      */
-    componentWillUnmount() {
-        this._clearAutoHideTimeout();
-    }
-
-    /**
-     * Implements React's {@link Component#render()}.
-     *
-     * @inheritdoc
-     * @returns {ReactElement}
-     */
-    render() {
-        if (this.state.hidden) {
-            return null;
-        }
-
-        const {
-            error: errorConstants,
-            mode: modeConstants,
-            status: statusConstants
-        } = JitsiRecordingConstants;
-        const { session } = this.props;
-        const allTranslationKeys = _getTranslationKeysByMode();
-        const translationKeys = allTranslationKeys[session.mode];
-        let circularLabelClass, circularLabelKey, messageKey;
-
-        switch (session.status) {
-        case statusConstants.OFF: {
-            if (session.error) {
-                messageKey = translationKeys.errors[session.error]
-                    || translationKeys.errors[errorConstants.ERROR];
-            } else {
-                messageKey = translationKeys.status[statusConstants.OFF];
-            }
-            break;
-        }
-        case statusConstants.ON:
-            circularLabelClass = session.mode;
-            circularLabelKey = session.mode === modeConstants.STREAM
-                ? 'recording.live' : 'recording.rec';
-            break;
-        case statusConstants.PENDING:
-            messageKey = translationKeys.status[statusConstants.PENDING];
-            break;
-        }
-
-        const className = `recording-label ${
```

Comment:
The amount of appreciation I have for you cleaning the label shenanigans is immeasurable. The label pulling double duty has been an issue for so long...and I've kicked the can down the road twice now, once for vertical filmstrip and once for simultaneously live streaming and recording. You're a better man than I.

Reply:
Nah, I just had enough time to work on it